File: Ejercicio2/Python/main.py
from enum import Enum
from classes.Grilla import Grilla
from classes.Movil import Movil,dirEnum
from classes.Autonomo import Autonomo
from random import randint

class randEnum(Enum):
    CAPACIDAD = 0
    CONSUMO = 1

def numeroAleatorio(tipo):
    if tipo == randEnum.CAPACIDAD.value:
        return randint(200, 300)
    elif tipo == randEnum.CONSUMO.value:
        return randint(20, 40)
    else:
        return 0

# Genero la grilla plana
ancho = int(input("Ingrese el ancho deseado de la grilla: "))
alto = int(input("Ingrese el alto deseado de la grilla: "))
grilla = Grilla(ancho, alto)
anchoGrilla = grilla.getAncho()
altoGrilla = grilla.getAlto()

# Generación del móvil manual
nombreMovilManual = input("\nIngrese el nombre del móvil manual: ")
movilManual = Movil(nombreMovilManual, anchoGrilla, altoGrilla)

# Movimientos del móvil manual
print("\nLa posición inicial de", movilManual.getNombre(), "es: (" + str(movilManual.getPosicionX()) + "," + str(movilManual.getPosicionY()) + ")\n")
seguirMoviendo = True
while seguirMoviendo:
    print("Ingrese la dirección de movimiento:")
    print(str(dirEnum.ADELANTE.value) + ": mover hacia ADELANTE")
    print(str(dirEnum.ATRAS.value) + ": mover hacia ATRÁS")
    print(str(dirEnum.DERECHA.value) + ": mover hacia la DERECHA")
    print(str(dirEnum.IZQUIERDA.value) + ": mover hacia la IZQUIERDA")
    print(str(dirEnum.TERMINAR.value) + ": terminar\n")
    movimiento = int(input("Movimiento: "))
    direccionMovimiento = movilManual.moverse(movimiento)
    print("")
    print(direccionMovimiento)
    print("Las coordenadas de la posición actual de", movilManual.getNombre(), "son: (" + str(movilManual.getPosicionX()) + "," + str(movilManual.getPosicionY()) + ")")
    print("Distancia recorrida hasta el momento:", movilManual.getDistanciaRecorrida(), "unidades\n")
    if (movimiento == dirEnum.TERMINAR.value):
        seguirMoviendo = False

print("----------------------------------------\n")

movilesAutonomos = []
for i in range(3):
    nombreMovilAutonomo = input("Ingresá el nombre del " + str(i+1) + "° móvil autónomo: ")
    print("")
    movilesAutonomos.append(Autonomo(nombreMovilAutonomo, anchoGrilla, altoGrilla, numeroAleatorio(randEnum.CAPACIDAD.value), numeroAleatorio(randEnum.CONSUMO.value)))
    print(movilesAutonomos[i].moverse())
    print("----------------------------------------\n")

File: Ejercicio2/C++/main.cpp
#include <iostream>
#include <ctime>
#include "classes/Grilla.h"
#include "classes/Movil.h"
#include "classes/Autonomo.h"
using namespace std;

enum numeroAleatorio {
    CAPACIDAD,
    CONSUMO
};

int numeroAleatorio(int tipo);

int main() {
    srand(time(NULL));
    //Genero la grilla plana
    int ancho, alto;
    cout << "Ingrese el ancho deseado de la grilla: ";
    cin >> ancho;
    cout << "Ingrese el alto deseado de la grilla: ";
    cin >> alto;
    cout << endl;
    Grilla *grilla = new Grilla(ancho, alto);
    int anchoGrilla = grilla ->getAncho();
    int altoGrilla = grilla->getAlto();

    // Generación del móvil manual
    string nombreMovilManual;
    cout << "Ingrese el nombre del móvil manual: ";
    cin >> nombreMovilManual;
    cout << endl;
    Movil* movilManual = new Movil(nombreMovilManual, anchoGrilla, altoGrilla);

    // Movimientos del móvil manual
    cout << "La posición inicial de " << movilManual->getNombre() << " es: (" << movilManual->getPosicionX() << "," << movilManual->getPosicionY() << ")" << endl << endl;
    int movimiento;
    string direccionMovimiento;
    bool seguirMoviendo = true;
    while (seguirMoviendo) {
        cout << "Ingrese la dirección de movimiento:" << endl
        << ADELANTE << ": mover hacia ADELANTE" << endl
        << ATRAS << ": mover hacia ATRÁS" << endl
        << DERECHA << ": mover hacia la DERECHA" << endl
        << IZQUIERDA << ": mover hacia la IZQUIERDA" << endl
        << TERMINAR << ": Terminar" << endl << endl;
        cout << "Movimiento: ";
        cin >> movimiento;
        direccionMovimiento = movilManual->moverse(movimiento);
        cout << endl << direccionMovimiento << endl;
        cout << "Las coordenadas de la posición actual de " << movilManual->getNombre() << " son: (" << movilManual->getPosicionX() << "," << movilManual->getPosicionY() << ")" << endl;
        cout << "Distancia recorrida hasta el momento: " << movilManual->getDistanciaRecorrida() << " unidades" << endl << endl;
        (movimiento == 0) && (seguirMoviendo = false);
    };

    cout << "----------------------------------------" << endl << endl;

    // Generación de los móviles autónomo
    Autonomo* movilesAutonomos[3];
    for (int i = 0; i < 3; i++) {
        string nombreMovilAutonomo;
        cout << "Ingrese el nombre del " << (i+1) << "° móvil autónomo: ";
        cin >> nombreMovilAutonomo;
        cout << endl;
        movilesAutonomos[i] = new Autonomo(nombreMovilAutonomo, anchoGrilla, altoGrilla, numeroAleatorio(CAPACIDAD), numeroAleatorio(CONSUMO));
        movilesAutonomos[i]->moverse();
        cout << "----------------------------------------" << endl << endl;
    }

    return 0;
};

int numeroAleatorio(int tipo) {
    switch (tipo) {
        case CAPACIDAD:
            return rand() % 100 + 200;
            break;
        case CONSUMO:
            return rand() % 20 + 20;
            break;
        default:
            return 0;
            break;
    };
};